stacker/providers/aws/default.py

```diff
@@ -409,6 +417,29 @@ def generate_cloudformation_args(stack_name, parameters, tags, template,
     return args
 
 
+class ProviderBuilder(object):
+    """Implements a ProviderBuilder for the AWS provider."""
+
+    def __init__(self, region=None, **kwargs):
+        self.region = region
+        self.kwargs = kwargs
+
+    def build(self, region=None):
+        # TODO(ejholmes): This class _could_ cache built providers, however,
+        # the building of boto3 clients is _not_ threadsafe. See
+        # https://github.com/boto/boto3/issues/801#issuecomment-245455979
+        if not region:
+            region = self.region
+        session = get_session(region=region)
+        return Provider(session, **self.kwargs)
+
+    def tail_stack(self, stack, cancel, retries=0, **kwargs):
```

Comment:
Good question. I'm definitely not happy with it, but it's a little tricky because of where we pass the [tail_stack](https://github.com/remind101/stacker/blob/b58e01c7281b925311bfad95a20fc68432b57635/stacker/actions/build.py#L329) function.

I could change that line to something like:

```python
tail=self._tail_stack
```

And then:

```python
def _tail_stack(...)
  provider = self.build_provider(stack)
  return provider.tail_stack(...)
```

Which would definitely be better, so we can share and re-use a `build_provider` method.